mcs/tools/gensources/gensources.cs

```diff
@@ -163,17 +162,27 @@ public static class Program {
             }
         }
 
-        TextWriter output;
-        if (useStdout)
-            output = Console.Out;
-        else
-            output = new StreamWriter (outFile);
+        int parts = files.Count == 0 ? 0 : files.Max (f => f.SplitNumber);
+        TextWriter[] outputs = new TextWriter[parts + 1];
 
-        using (output) {
-            foreach (var fileName in fileNames)
-                output.WriteLine (fileName);
+        for (int i = 0; i < parts + 1; i++) {
+            outputs[i] = useStdout ? Console.Out : new StreamWriter (i == 0 ? outFile : $"{outFile}.part{i}");
         }
 
+        foreach (var fileName in files) {
+            if (fileName.SplitNumber == 0) {
```

Comment:
I'd like to see SplitNumber be int? with the ```null``` case being special instead of 0, or a comment somewhere explaining the meaning of 0 vs other values

Reply:
Added a comment.